packages/govern-subgraph/src/utils/events.ts

```diff
@@ -1,42 +1,164 @@
-import { BigInt } from '@graphprotocol/graph-ts'
+import { BigInt, ethereum } from '@graphprotocol/graph-ts'
 import {
+  Collateral as CollateralEntity,
   Container as ContainerEntity,
-  ContainerEvent as ContainerEventEntity
+  ContainerEventChallenge as ContainerEventChallengeEntity,
+  ContainerEventExecute as ContainerEventExecuteEntity,
+  ContainerEventResolve as ContainerEventResolveEntity,
+  ContainerEventRule as ContainerEventRuleEntity,
+  ContainerEventSchedule as ContainerEventScheduleEntity,
+  ContainerEventSubmitEvidence as ContainerEventSubmitEvidenceEntity,
+  ContainerEventVeto as ContainerEventVetoEntity,
 } from '../../generated/schema'
-import { buildIndexedId } from './ids'
+import { Executed as ExecutedEvent } from '../../generated/templates/Govern/Govern'
+import {
+  Challenged as ChallengedEvent,
+  Resolved as ResolvedEvent,
+  Ruled as RuledEvent,
+  Scheduled as ScheduledEvent,
+  EvidenceSubmitted as EvidenceSubmittedEvent,
+  Vetoed as VetoedEvent,
+} from '../../generated/templates/GovernQueue/GovernQueue'
+import { buildId, buildIndexedId } from './ids'
+
+function finalizeContainerEvent<T, U>(
+  container: ContainerEntity,
+  containerEvent: U,
+  ethereumEvent: T
+): U {
+  containerEvent.createdAt = ethereumEvent.block.timestamp
+  container.history.push(containerEvent.id)
```

Comment:
I think if we only push the new entity here the `Container.hisotry` is not updated on the graph-node. Instead we should:
```
	let containerEventsHistory = container.history
	containerEventsHistory(containerEvent.id)
	container.history = containerEventsHistory
```

Reply:
Was trying to find the piece of documentation where they explain this but I couldn't find it.

Reply:
Oooh that’s why. I wonder if we could have a utility function to make the intention clear? Will look into it.

Reply:
Heheh yeah this is a bit tricky... I honestly wonder if this is a Assemblyscript limitation or just something that's been overlooked. 😅

Reply:
I was reading about it, it is due to the type conversion happening in the schema:

![image](https://user-images.githubusercontent.com/36158/97482095-994bf400-194d-11eb-8950-6f2701c2cf3c.png)

This is how arrays get handled in the generated code, so accessing them in any way creates a copy rather than returning a reference.

Reply:
Thanks a lot for sharing!